File: poh4/poh4_3.scala
//usr/bin/env scala $0 $@;exit
object Main{
	def main(args:Array[String]){
		val Array(_T,_N) = readLine().split(" ").map(_.toInt)
		var cur=0
		var i=0
		val a=new Array[Int](_N)
		while(i<_T){
			a(i)=readInt()
			cur+=a(i)
			i+=1
		}
		var r=cur
		while(i<_N){
			a(i)=readInt()
			cur+=a(i)
			cur-=a(i-_T)
			if(r<cur)r=cur
			i+=1
		}
		println(r)
	}
}

File: poh4/poh4_3.swift
//usr/bin/env swift $0 $@;exit

let _a=readLine()!.characters.split{$0==" "}
let T=Int(String(_a[0]))!,N=Int(String(_a[1]))!
var a=[Int](repeating:0,count:N)
var cur=0,i=0

for i in 0..<T {
	a[i]=Int(readLine()!)!
	cur+=a[i]
}
var r=cur
for i in T..<N {
	a[i]=Int(readLine()!)!
	cur+=a[i]
	cur-=a[i-T]
	if r<cur {r=cur}
}
print(r)
